### ISSUE TITLE
`BoundAnonymousObjectCreationExpression` in CSharp is exposed as a `None` operation

### ISSUE THREAD

**Issue by balajikris** [dotnet/roslyn#8511]
Similar to #8476 but different stack.
1. File -> New Project -> C# -> Web -> MVC
2. Wait for project to load.

**Result**: VS crashes.

```
Application: devenv.exe
Framework Version: v4.0.30319
Description: The application requested process termination through System.Environment.FailFast(string message).
Message: System.InvalidOperationException: This program location is thought to be unreachable.
   at Microsoft.CodeAnalysis.CSharp.BoundAnonymousObjectCreationExpression.Accept(OperationVisitor visitor)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.Visit(IOperation operation)
   at Microsoft.CodeAnalysis.Semantics.OperationExtensions.OperationCollector.Visit(IOperation operation)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.VisitConversionExpression(IConversionExpression operation)
   at Microsoft.CodeAnalysis.CSharp.BoundConversion.Accept(OperationVisitor visitor)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.Visit(IOperation operation)
   at Microsoft.CodeAnalysis.Semantics.OperationExtensions.OperationCollector.Visit(IOperation operation)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.VisitArgument(IArgument operation)
   at Microsoft.CodeAnalysis.CSharp.BoundCall.ArgumentBase.Microsoft.CodeAnalysis.Semantics.IOperation.Accept(OperationVisitor visitor)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.Visit(IOperation operation)
   at Microsoft.CodeAnalysis.Semantics.OperationExtensions.OperationCollector.Visit(IOperation operation)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.VisitArray[T](ImmutableArray`1 list)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.VisitInvocationExpression(IInvocationExpression operation)
   at Microsoft.CodeAnalysis.CSharp.BoundCall.Accept(OperationVisitor visitor)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.Visit(IOperation operation)
   at Microsoft.CodeAnalysis.Semantics.OperationExtensions.OperationCollector.Visit(IOperation operation)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.VisitExpressionStatement(IExpressionStatement operation)
   at Microsoft.CodeAnalysis.CSharp.BoundExpressionStatement.Accept(OperationVisitor visitor)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.Visit(IOperation operation)
   at Microsoft.CodeAnalysis.Semantics.OperationExtensions.OperationCollector.Visit(IOperation operation)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.VisitArray[T](ImmutableArray`1 list)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.VisitBlockStatement(IBlockStatement operation)
   at Microsoft.CodeAnalysis.CSharp.BoundBlock.Accept(OperationVisitor visitor)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.Visit(IOperation operation)
   at Microsoft.CodeAnalysis.Semantics.OperationExtensions.OperationCollector.Visit(IOperation operation)
   at Microsoft.CodeAnalysis.Semantics.OperationExtensions.DescendantsAndSelf(IOperation operation)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver`1.GetOperationsToAnalyze(ImmutableArray`1 operationBlocks)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver`1.ExecuteDeclaringReferenceActions(SyntaxReference decl, Int32 declarationIndex, SymbolDeclaredCompilationEvent symbolEvent, AnalysisScope analysisScope, AnalysisState analysisStateOpt, Boolean shouldExecuteSyntaxNodeActions, Boolean shouldExecuteOperationActions, Boolean shouldExecuteCodeBlockActions, Boolean shouldExecuteOperationBlockActions, Boolean isInGeneratedCode, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver`1.ExecuteDeclaringReferenceActions(SymbolDeclaredCompilationEvent symbolEvent, AnalysisScope analysisScope, AnalysisState analysisStateOpt, Boolean isGeneratedCodeSymbol, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.ProcessSymbolDeclared(SymbolDeclaredCompilationEvent symbolEvent, AnalysisScope analysisScope, AnalysisState analysisStateOpt, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.ProcessEventCore(CompilationEvent e, AnalysisScope analysisScope, AnalysisState analysisStateOpt, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.ProcessEvent(CompilationEvent e, AnalysisScope analysisScope, AnalysisState analysisStateOpt, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.<ProcessCompilationEventsCoreAsync>d__64.MoveNext()
Stack:
   at System.Environment.FailFast(System.String, System.Exception)
   at Microsoft.CodeAnalysis.FailFast.OnFatalException(System.Exception)
   at Microsoft.CodeAnalysis.FatalError.Report(System.Exception, System.Action`1<System.Exception>)
   at Microsoft.CodeAnalysis.FatalError.Report(System.Exception)
   at Microsoft.CodeAnalysis.FatalError.ReportUnlessCanceled(System.Exception)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver+<ProcessCompilationEventsCoreAsync>d__64.MoveNext()
   at Microsoft.CodeAnalysis.CSharp.BoundAnonymousObjectCreationExpression.Accept(Microsoft.CodeAnalysis.Semantics.OperationVisitor)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.Visit(Microsoft.CodeAnalysis.Semantics.IOperation)
   at Microsoft.CodeAnalysis.Semantics.OperationExtensions+OperationCollector.Visit(Microsoft.CodeAnalysis.Semantics.IOperation)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.VisitConversionExpression(Microsoft.CodeAnalysis.Semantics.IConversionExpression)
   at Microsoft.CodeAnalysis.CSharp.BoundConversion.Accept(Microsoft.CodeAnalysis.Semantics.OperationVisitor)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.Visit(Microsoft.CodeAnalysis.Semantics.IOperation)
   at Microsoft.CodeAnalysis.Semantics.OperationExtensions+OperationCollector.Visit(Microsoft.CodeAnalysis.Semantics.IOperation)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.VisitArgument(Microsoft.CodeAnalysis.Semantics.IArgument)
   at Microsoft.CodeAnalysis.CSharp.BoundCall+ArgumentBase.Microsoft.CodeAnalysis.Semantics.IOperation.Accept(Microsoft.CodeAnalysis.Semantics.OperationVisitor)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.Visit(Microsoft.CodeAnalysis.Semantics.IOperation)
   at Microsoft.CodeAnalysis.Semantics.OperationExtensions+OperationCollector.Visit(Microsoft.CodeAnalysis.Semantics.IOperation)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.VisitArray[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]](System.Collections.Immutable.ImmutableArray`1<System.__Canon>)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.VisitInvocationExpression(Microsoft.CodeAnalysis.Semantics.IInvocationExpression)
   at Microsoft.CodeAnalysis.CSharp.BoundCall.Accept(Microsoft.CodeAnalysis.Semantics.OperationVisitor)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.Visit(Microsoft.CodeAnalysis.Semantics.IOperation)
   at Microsoft.CodeAnalysis.Semantics.OperationExtensions+OperationCollector.Visit(Microsoft.CodeAnalysis.Semantics.IOperation)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.VisitExpressionStatement(Microsoft.CodeAnalysis.Semantics.IExpressionStatement)
   at Microsoft.CodeAnalysis.CSharp.BoundExpressionStatement.Accept(Microsoft.CodeAnalysis.Semantics.OperationVisitor)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.Visit(Microsoft.CodeAnalysis.Semantics.IOperation)
   at Microsoft.CodeAnalysis.Semantics.OperationExtensions+OperationCollector.Visit(Microsoft.CodeAnalysis.Semantics.IOperation)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.VisitArray[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]](System.Collections.Immutable.ImmutableArray`1<System.__Canon>)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.VisitBlockStatement(Microsoft.CodeAnalysis.Semantics.IBlockStatement)
   at Microsoft.CodeAnalysis.CSharp.BoundBlock.Accept(Microsoft.CodeAnalysis.Semantics.OperationVisitor)
   at Microsoft.CodeAnalysis.Semantics.OperationWalker.Visit(Microsoft.CodeAnalysis.Semantics.IOperation)
   at Microsoft.CodeAnalysis.Semantics.OperationExtensions+OperationCollector.Visit(Microsoft.CodeAnalysis.Semantics.IOperation)
   at Microsoft.CodeAnalysis.Semantics.OperationExtensions.DescendantsAndSelf(Microsoft.CodeAnalysis.Semantics.IOperation)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver`1[[Microsoft.CodeAnalysis.CSharp.SyntaxKind, Microsoft.CodeAnalysis.CSharp, Version=42.42.42.42, Culture=neutral, PublicKeyToken=31bf3856ad364e35]].GetOperationsToAnalyze(System.Collections.Immutable.ImmutableArray`1<Microsoft.CodeAnalysis.Semantics.IOperation>)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver`1[[Microsoft.CodeAnalysis.CSharp.SyntaxKind, Microsoft.CodeAnalysis.CSharp, Version=42.42.42.42, Culture=neutral, PublicKeyToken=31bf3856ad364e35]].ExecuteDeclaringReferenceActions(Microsoft.CodeAnalysis.SyntaxReference, Int32, Microsoft.CodeAnalysis.Diagnostics.SymbolDeclaredCompilationEvent, Microsoft.CodeAnalysis.Diagnostics.AnalysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalysisState, Boolean, Boolean, Boolean, Boolean, Boolean, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver`1[[Microsoft.CodeAnalysis.CSharp.SyntaxKind, Microsoft.CodeAnalysis.CSharp, Version=42.42.42.42, Culture=neutral, PublicKeyToken=31bf3856ad364e35]].ExecuteDeclaringReferenceActions(Microsoft.CodeAnalysis.Diagnostics.SymbolDeclaredCompilationEvent, Microsoft.CodeAnalysis.Diagnostics.AnalysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalysisState, Boolean, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.ProcessSymbolDeclared(Microsoft.CodeAnalysis.Diagnostics.SymbolDeclaredCompilationEvent, Microsoft.CodeAnalysis.Diagnostics.AnalysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalysisState, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.ProcessEventCore(Microsoft.CodeAnalysis.Diagnostics.CompilationEvent, Microsoft.CodeAnalysis.Diagnostics.AnalysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalysisState, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.ProcessEvent(Microsoft.CodeAnalysis.Diagnostics.CompilationEvent, Microsoft.CodeAnalysis.Diagnostics.AnalysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalysisState, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver+<ProcessCompilationEventsCoreAsync>d__64.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].Start[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]](System.__Canon ByRef)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.ProcessCompilationEventsCoreAsync(Microsoft.CodeAnalysis.Diagnostics.AnalysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalysisState, Boolean, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver+<ProcessCompilationEventsAsync>d__63.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]](System.__Canon ByRef)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.ProcessCompilationEventsAsync(Microsoft.CodeAnalysis.Diagnostics.AnalysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalysisState, Boolean, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver+<ExecutePrimaryAnalysisTaskAsync>d__42.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]](System.__Canon ByRef)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.ExecutePrimaryAnalysisTaskAsync(Microsoft.CodeAnalysis.Diagnostics.AnalysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalysisState, Boolean, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver+<AttachQueueAndProcessAllEventsAsync>d__40.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]](System.__Canon ByRef)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.AttachQueueAndProcessAllEventsAsync(Microsoft.CodeAnalysis.Diagnostics.AsyncQueue`1<Microsoft.CodeAnalysis.Diagnostics.CompilationEvent>, Microsoft.CodeAnalysis.Diagnostics.AnalysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalysisState, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzers+<ComputeAnalyzerDiagnosticsCoreAsync>d__55.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]](System.__Canon ByRef)
   at Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzers.ComputeAnalyzerDiagnosticsCoreAsync(Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver, Microsoft.CodeAnalysis.Diagnostics.AsyncQueue`1<Microsoft.CodeAnalysis.Diagnostics.CompilationEvent>, Microsoft.CodeAnalysis.Diagnostics.AnalysisScope, System.Threading.CancellationToken)
   at Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzers+<>c__DisplayClass52_2+<<ComputeAnalyzerDiagnosticsAsync>b__1>d.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]](System.__Canon ByRef)
   at Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzers+<>c__DisplayClass52_2.<ComputeAnalyzerDiagnosticsAsync>b__1()
   at System.Threading.Tasks.Task`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
   at System.Threading.Tasks.Task.ExecutionContextCallback(System.Object)
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(System.Threading.Tasks.Task ByRef)
   at System.Threading.Tasks.Task.ExecuteEntry(Boolean)
   at System.Threading.Tasks.Task.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()

```

**Comment by balajikris** [comment]
looks like i got the wrong stack, will update.

**Comment by davkean** [comment]
@srivatsn Updated the urgency - this is going to fail DDRITs next time we insert if above is actually the repro.

**Comment by genlu** [comment]
So this is definitely fixed by @8517, right @balajikris ?
By the way, you will need an analyzer installed to verify this.

**Comment by balajikris** [comment]
@genlu - Please feel free to test and close. My VS hives are not stable and I do not have enough context about installing analyzers either. Also, we already know from your testing that it looks fixed :) I'm okay if you close this bug via #8517.

**Comment by genlu** [comment]
Removed the urgency-now label as the crash is fixed by #8517. I will keep using this issue to track the underlying problem in IOperation.

@JohnHamby Is this related to generated bound nodes? Could you please take a look at it?

**Comment by genlu** [comment]
This is a dup of #19926, and should be fixed by #20364.
